### PR TITLE
Add "watch" actions which run `tsc` in watch mode

### PR DESCRIPTION
**Pull request by JonathanDCohen**
"watch" mode watches for any changed files or directories, recompiling anything which has been changed without needing a full rebuild.  This can be used during development to speed up rebuild times and edit-compile iteration.

### CODE REVIEW

File: src/electron/watch_action.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Builds the project in watch mode, recompiling TS files on change.
+# See https://www.typescriptlang.org/docs/handbook/compiler-options.html
+
+yarn do src/www/build
```
Comment (fortuna) [inline src/electron/watch_action.sh]: Maybe this should be `yarn do src/www/watch&` ?

**Review by fortuna — DISMISSED** [review]

**Comment by daniellacosse** [comment]
@fortuna can we merge this? it'd be very useful for me

also, i think this outmodes https://github.com/Jigsaw-Code/outline-client/pull/759 and https://github.com/Jigsaw-Code/outline-client/pull/534

**Review by fortuna — APPROVED** [review]

**Review by fortuna — CHANGES_REQUESTED** [review]
Wait

**Comment by fortuna** [comment]
@daniellacosse Setting up the dev server will be a lot more productive. Also, we use webpack now. The change to webpack happened after this PR. We shouldn't be calling tsc directly.

**Comment by fortuna** [comment]
See Outline Manager for how we set up the dev server.

**Comment by daniellacosse** [comment]
> @daniellacosse Setting up the dev server will be a lot more productive. Also, we use webpack now. The change to webpack happened after this PR. We shouldn't be calling tsc directly.

We should close those other issues then, yeah?  https://github.com/Jigsaw-Code/outline-client/pull/759 and https://github.com/Jigsaw-Code/outline-client/pull/534?

**Comment by fortuna** [comment]
No, closing people bugs and PRs is usually bad. We can do that once we add the dev server.

**Comment by daniellacosse** [comment]
> No, closing people bugs and PRs is usually bad. We can do that once we add the dev server.

Ah, right, though my understanding is correct

**Pull request closed without merging**